### DISCUSSION TITLE
Call internal controller from server side

### DISCUSSION THREAD

**Discussion by rhemasystems** [serenity-is/Serenity#6773] [Help]
I have the below code inside AfterSave handler method, it works fine but the base URL is hardcoded.
How can I set the baseURL without hardcoding it as the API is also in the solution. Thank you

`var client = new JsonServiceClient("http://localhost:63090/");`
 `client.Post<ServiceResponse>("LeaveRequestEmail/SendEmail",`
                    `   new LeaveRequestModelESS`
                   `    {`
                        `       LeaveNumber = Row.Number,`
                        `       LeaveRequestStatus = (Int16)Row.LeaveRequestStatus,`
                       `        RequesterApprover = 1, //ALways be a requester`
                        `       LeaveRequestRow = leaveinfo,`
                    `     });`

**Comment by LSNicholls** [comment]
If you have access to an object implementing IConfiguration, you can get this out of appsettings.json, right?  This would be EnvironmentSettings:SiteExternalUrl?

Alternatively, you can store this in a SQL table and query it from there.

Actually, there's probably a better way that somebody else will tell you (I am far from an expert and look forward to hearing about the "right" Serenity way!).  But I know that either of these suggestions *would* work.
